Sources/NIOHPACK/HPACKHeader.swift

```diff
@@ -416,6 +419,22 @@ extension HPACKHeaders: Equatable {
     }
 }
 
+extension HPACKHeaders: Hashable {
+    @inlinable
+    public func hash(into hasher: inout Hasher) {
+        // Discriminator, to indicate that this is a collection. This improves the performance
+        // of Sets and Dictionaries that include collections of HPACKHeaders by reducing hash collisions.
+        hasher.combine(self.count)
+
+        // This emulates the logic used in equatability, but we sort it to ensure that
+        // we hash equivalently.
+        let names = Set(self.headers.map { $0.name }).sorted()
```

Comment:
Why do we need to load into a `Set` to then immediately extract?

Reply:
Flattening uniqueness: it's possible to have multiple entries with the same name. This is what happens above in the equatable implementation.

Reply:
We can remove one allocation if we lazily map over the headers to extract the name
```suggestion
        let names = Set(self.headers.lazy.map { $0.name }).sorted()
```